src/ng/compile.js

```diff
@@ -2592,11 +2587,13 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             break;
 
           case '=':
-            if (optional && !attrs[attrName]) {
-              return;
+            if (!hasOwnProperty.call(attrs, attrName)) {
```

Comment:
The code in that snippet is contained in the second commit here, just organized slightly differently --- I decided to avoid the `isString()` check and just test if it's falsy (values in `attrs` should always be a string, unless it gets shadowed above, in which case it's undefined --- so also falsy) --- so the simpler version in this PR should be fine